src/cmd/sign.js

```diff
@@ -107,6 +117,27 @@ export default function sign(
         downloadDir: artifactsDir,
       });
 
+      if (updateLink) {
+        let extensionID;
+        if (signingResult.id) {
+          extensionID = signingResult.id;
+        } else {
+          extensionID = id;
+        }
+        // flow thinks that oldUpdateManifestData can be potentially undefined
+        // this is wrong however, the if-case guarding both the `oldUpdateManifestData = `
+        // statement and this function call are both checking
+        // the exact same thing, which doesn't change
+        // anywhere in the code. Basically, `if (A) then define B` and later on `if (A) then use B`
+        // which flow doesn't seem to understand
+        let oldUpdateManifestDataMakeFlowHappy =
+          oldUpdateManifestData == null ? {} : oldUpdateManifestData;
```

Comment:
@kumar303 Ah yeah, I didn't consider the shadowing. Still, the flow problem persists.